test failure in 1.18.3

Hello,

I can reproduce a test failure in a RockyLinux-9 docker container.

Elixir version: 1.18.3
Erlang version: 27.3.3

This is the output of `make -j2 && make -j2 test`:
```
#7 1641.6 Recompile: src/iex
#7 1641.9 Recompile: src/elixir_utils
#7 1642.2 Recompile: src/elixir_tokenizer
#7 1644.8 Recompile: src/elixir_sup
#7 1644.9 Recompile: src/elixir_rewrite
#7 1645.4 Recompile: src/elixir_quote
#7 1645.9 Recompile: src/elixir_parser
#7 1657.7 Recompile: src/elixir_overridable
#7 1657.8 Recompile: src/elixir_module
#7 1658.4 Recompile: src/elixir_map
#7 1658.6 Recompile: src/elixir_lexical
#7 1658.7 Recompile: src/elixir_json
#7 1664.1 Recompile: src/elixir_interpolation
#7 1664.7 Recompile: src/elixir_import
#7 1664.9 Recompile: src/elixir_fn
#7 1665.1 Recompile: src/elixir_expand
#7 1666.3 Recompile: src/elixir_errors
#7 1666.6 Recompile: src/elixir_erl_var
#7 1666.7 Recompile: src/elixir_erl_try
#7 1666.9 Recompile: src/elixir_erl_pass
#7 1667.7 Recompile: src/elixir_erl_for
#7 1668.1 Recompile: src/elixir_erl_compiler
#7 1668.2 Recompile: src/elixir_erl_clauses
#7 1668.3 Recompile: src/elixir_erl
#7 1668.9 Recompile: src/elixir_env
#7 1669.0 Recompile: src/elixir_dispatch
#7 1669.3 Recompile: src/elixir_def
#7 1669.7 Recompile: src/elixir_config
#7 1669.8 Recompile: src/elixir_compiler
#7 1669.9 Recompile: src/elixir_code_server
#7 1670.0 Recompile: src/elixir_clauses
#7 1670.4 Recompile: src/elixir_bootstrap
#7 1670.4 Recompile: src/elixir_bitstring
#7 1670.7 Recompile: src/elixir_aliases
#7 1670.8 Recompile: src/elixir
#7 1672.0 Generated elixir app
#7 1672.0 ==> bootstrap (compile)
#7 1674.9 Compiled kernel.ex
#7 1675.2 Compiled kernel/utils.ex
#7 1675.5 Compiled macro/env.ex
#7 1676.0 Compiled keyword.ex
#7 1677.0 Compiled module.ex
#7 1677.4 Compiled list.ex
#7 1679.1 Compiled macro.ex
#7 1680.6 Compiled kernel/typespec.ex
#7 1681.1 Compiled code.ex
#7 1681.3 Compiled code/identifier.ex
#7 1681.9 Compiled protocol.ex
#7 1682.2 Compiled stream/reducers.ex
#7 1685.3 Compiled enum.ex
#7 1685.8 Compiled regex.ex
#7 1686.6 Compiled inspect/algebra.ex
#7 1687.7 Compiled inspect.ex
#7 1689.1 Compiled string.ex
#7 1689.3 Compiled string/chars.ex
#7 1691.8 Compiled kernel.ex
#7 1692.0 Compiled list/chars.ex
#7 1692.1 Compiled bitwise.ex
#7 1692.6 Compiled module/parallel_checker.ex
#7 1693.0 Compiled module/behaviour.ex
#7 1693.2 Compiled module/types/helpers.ex
#7 1695.4 Compiled module/types/descr.ex
#7 1695.8 Compiled module/types/of.ex
#7 1696.7 Compiled module/types/pattern.ex
#7 1698.0 Compiled module/types/apply.ex
#7 1699.1 Compiled module/types/expr.ex
#7 1699.5 Compiled module/types.ex
#7 1702.3 Compiled exception.ex
#7 1702.7 Compiled path.ex
#7 1703.4 Compiled file.ex
#7 1703.7 Compiled map.ex
#7 1703.7 Compiled function.ex
#7 1704.1 Compiled range.ex
#7 1704.7 Compiled access.ex
#7 1705.0 Compiled io.ex
#7 1705.6 Compiled system.ex
#7 1707.9 Compiled code/formatter.ex
#7 1708.5 Compiled code/normalizer.ex
#7 1709.2 Compiled kernel/cli.ex
#7 1709.3 Compiled kernel/error_handler.ex
#7 1710.2 Compiled kernel/parallel_compiler.ex
#7 1710.4 Compiled kernel/lexical_tracker.ex
#7 1710.4 make[1]: Entering directory '/b/elixir-1.18.3'
#7 1710.4 ==> unicode (compile)
#7 1714.8 [Unicode] Break on 25 whitespace codepoints
#7 1716.4 [Unicode] Tokenizing 112700 non-ascii codepoints
#7 1716.8 [Unicode] Tokenizing 65 scriptsets
#7 1723.4 make[1]: Leaving directory '/b/elixir-1.18.3'
#7 1723.4 ==> elixir (compile)
#7 1754.9 Generated elixir app
#7 1754.9 ==> eex (compile)
#7 1756.9 ==> mix (compile)
#7 1767.3 Generated mix app
#7 1767.3 ==> ex_unit (compile)
#7 1773.3 Generated ex_unit app
#7 1773.3 ==> logger (compile)
#7 1777.0 Generated logger app
#7 1778.2 Generated eex app
#7 1778.2 ==> iex (compile)
#7 1783.1 Generated iex app
#7 1794.4 ==> elixir (eunit)
#7 1795.6   All 126 tests passed.
#7 1795.6 
#7 1795.6 ==> elixir (ex_unit)
#7 1858.4 Running ExUnit with seed: 739527, max_cases: 4
#7 1858.4 Excluding tags: [windows: true]
#7 1858.4 
#7 1858.4 .................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
#7 1920.7 Finished in 123.1 seconds (60.8s on load, 45.9s async, 16.4s sync)
#7 1920.7 2040 doctests, 4550 tests, 0 failures, 13 excluded
#7 1920.7 ==> ex_unit (ex_unit)
#7 1929.5 Running ExUnit with seed: 847311, max_cases: 4
#7 1929.5 
#7 1929.6 ........................................................................................................................................................................................................................................................................................................................................................................................................................................................
#7 1933.8 Finished in 11.8 seconds (7.6s on load, 0.6s async, 3.6s sync)
#7 1933.8 48 doctests, 392 tests, 0 failures
#7 1933.9 ==> logger (ex_unit)
#7 1939.4 Running ExUnit with seed: 755924, max_cases: 4
#7 1939.4 
#7 1939.4 ................................................................................................................................................................
#7 1940.4 Finished in 5.1 seconds (4.1s on load, 0.04s async, 0.9s sync)
#7 1940.4 3 doctests, 157 tests, 0 failures
#7 1940.4 ==> eex (ex_unit)
#7 1942.4 Running ExUnit with seed: 753143, max_cases: 4
#7 1942.4 
#7 1942.4 .................................................................................................................
#7 1942.5 Finished in 0.9 seconds (0.8s on load, 0.1s async, 0.00s sync)
#7 1942.5 7 doctests, 106 tests, 0 failures
#7 1942.6 ==> iex (ex_unit)
#7 1946.5 Running ExUnit with seed: 871427, max_cases: 4
#7 1946.5 
#7 1946.5 .........................................................................................................................................................................................................................................................................
#7 1963.3 Finished in 19.5 seconds (2.7s on load, 0.7s async, 16.0s sync)
#7 1963.3 265 tests, 0 failures
#7 1963.4 ==> mix (ex_unit)
#7 1979.1 Running ExUnit with seed: 340699, max_cases: 4
#7 1979.1 Excluding tags: [windows: true]
#7 1979.1 
#7 1979.2 .
#7 1979.3 
#7 1979.3   1) test delivers broadcast to subscribers for different keys (Mix.Sync.PubSubTest)
#7 1979.3      /b/elixir-1.18.3/lib/mix/test/mix/sync/pubsub_test.exs:39
#7 1979.3      Assertion failed, no matching message after 100ms
#7 1979.3      The process mailbox is empty.
#7 1979.3      code: assert_receive :subscribed1
#7 1979.3      stacktrace:
#7 1979.3        test/mix/sync/pubsub_test.exs:60: (test)
#7 1979.3 
#7 1979.3 .............................................................................................................................................................................................................................................................................................................    warning: redefining module Mix.Tasks.Local.Sample (current version defined in memory)
#7 2038.6     │
#7 2038.6   1 │ defmodule Mix.Tasks.Local.Sample do
#7 2038.6     │ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
#7 2038.6     │
#7 2038.6     └─ lib/local.sample.ex:1: Mix.Tasks.Local.Sample (module)
#7 2038.6 
#7 2038.6 ................................
#7 2044.7 09:17:03.156 [notice] Application parsetools exited: :stopped
#7 2044.7 .
#7 2044.8 09:17:03.200 [notice] Application parsetools exited: :stopped
#7 2044.8 .
#7 2044.8 09:17:03.239 [notice] Application parsetools exited: :stopped
#7 2044.8 .....
#7 2047.1 09:17:05.523 [notice] Application parsetools exited: :stopped
#7 2047.1 ..
#7 2047.2 09:17:05.573 [notice] Application parsetools exited: :stopped
#7 2047.2 
#7 2047.2 09:17:05.622 [notice] Application parsetools exited: :stopped
#7 2047.2 .
#7 2047.3 09:17:05.688 [notice] Application parsetools exited: :stopped
#7 2047.3 ....................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................................
#7 2408.6 Finished in 443.6 seconds (14.2s on load, 0.5s async, 428.9s sync)
#7 2408.6 6 doctests, 855 tests, 1 failure
#7 2408.6 make: *** [Makefile:116: test_mix] Error 2
```

Is this a know issue?
Is there something else I can provide to help debug the issue?

Thanks

--- a/lib/elixir/test/elixir/process_test.exs
+++ b/lib/elixir/test/elixir/process_test.exs
@@ -63,7 +63,7 @@ defmodule ProcessTest do
 
   test "sleep/1 with 2^32" do
     {pid, monitor_ref} = spawn_monitor(fn -> Process.sleep(2 ** 32) end)
-    refute_receive {:DOWN, ^monitor_ref, :process, ^pid, {:timeout_value, _trace}}
+    refute_receive {:DOWN, ^monitor_ref, :process, ^pid, {:timeout_value, _trace}}, 100
     Process.exit(pid, :kill)
   end
 
@@ -130,7 +130,7 @@ defmodule ProcessTest do
       end)
 
     true = Process.exit(pid, :normal)
-    refute_receive {:EXIT, ^pid, :normal}
+    refute_receive {:EXIT, ^pid, :normal}, 100
     assert Process.alive?(pid)
 
     # now exit the process for real so it doesn't hang around

--- a/lib/elixir/test/elixir/stream_test.exs
+++ b/lib/elixir/test/elixir/stream_test.exs
@@ -318,14 +318,14 @@ defmodule StreamTest do
     send(pid, {:stream, 1})
     send(pid, {:stream, 2})
     send(pid, {:stream, 3})
-    refute_receive {:stream, 1}
+    refute_receive {:stream, 1}, 100
 
     send(pid, {:stream, 4})
     assert_receive {:stream, 1}
 
     send(pid, {:stream, 5})
     assert_receive {:stream, 2}
-    refute_receive {:stream, 3}
+    refute_receive {:stream, 3}, 100
   end
 
   test "drop_every/2" do

--- a/lib/elixir/test/elixir/test_helper.exs
+++ b/lib/elixir/test/elixir/test_helper.exs
@@ -100,7 +100,6 @@ defmodule CodeFormatterHelpers do
   end
 end
 
-assert_timeout = String.to_integer(System.get_env("ELIXIR_ASSERT_TIMEOUT") || "500")
 epmd_exclude = if match?({:win32, _}, :os.type()), do: [epmd: true], else: []
 os_exclude = if PathHelpers.windows?(), do: [unix: true], else: [windows: true]
 
@@ -135,9 +134,9 @@ cover_exclude =
 
 ExUnit.start(
   trace: !!System.get_env("TRACE"),
-  assert_receive_timeout: assert_timeout,
   exclude:
     epmd_exclude ++
       os_exclude ++ line_exclude ++ distributed_exclude ++ source_exclude ++ cover_exclude,
-  include: line_include
+  include: line_include,
+  assert_receive_timeout: String.to_integer(System.get_env("ELIXIR_ASSERT_TIMEOUT", "300"))
 )

--- a/lib/ex_unit/examples/one_of_each.exs
+++ b/lib/ex_unit/examples/one_of_each.exs
@@ -104,7 +104,7 @@ defmodule TestOneOfEach do
 
   test "19. refute a message is received within a timeout" do
     send(self(), {:hello, "Dave"})
-    refute_receive {:hello, _}, 1000
+    refute_receive {:hello, _}, 100
   end
 
   test "20. refute a message is ready to be received" do

--- a/lib/ex_unit/test/ex_unit/assertions_test.exs
+++ b/lib/ex_unit/test/ex_unit/assertions_test.exs
@@ -601,7 +601,7 @@ defmodule ExUnit.AssertionsTest do
   end
 
   test "refute receive waits" do
-    false = refute_receive :hello
+    false = refute_receive :hello, 100
   end
 
   test "refute received when equal" do

--- a/lib/ex_unit/test/ex_unit/capture_log_test.exs
+++ b/lib/ex_unit/test/ex_unit/capture_log_test.exs
@@ -102,7 +102,7 @@ defmodule ExUnit.CaptureLogTest do
       Logger.error("oh no!")
     end)
 
-    refute_receive {:EXIT, _, _}
+    refute_receive {:EXIT, _, _}, 100
   end
 
   describe "with_log/2" do

--- a/lib/ex_unit/test/test_helper.exs
+++ b/lib/ex_unit/test/test_helper.exs
@@ -13,5 +13,6 @@ CoverageRecorder.maybe_record("ex_unit")
 ExUnit.start(
   trace: !!System.get_env("TRACE"),
   include: line_include,
-  exclude: line_exclude
+  exclude: line_exclude,
+  assert_receive_timeout: String.to_integer(System.get_env("ELIXIR_ASSERT_TIMEOUT", "300"))
 )

--- a/lib/iex/test/test_helper.exs
+++ b/lib/iex/test/test_helper.exs
@@ -7,8 +7,6 @@ path = Path.expand("../tmp/beams", __DIR__)
 File.rm_rf!(path)
 File.mkdir_p!(path)
 Code.prepend_path(path)
-
-assert_timeout = String.to_integer(System.get_env("ELIXIR_ASSERT_TIMEOUT") || "500")
 System.put_env("ELIXIR_EDITOR", "echo")
 
 {:ok, _} = Application.ensure_all_started(:iex)
@@ -42,10 +40,10 @@ cover_exclude =
   end
 
 ExUnit.start(
-  assert_receive_timeout: assert_timeout,
   trace: !!System.get_env("TRACE"),
   include: line_include,
-  exclude: line_exclude ++ erlang_doc_exclude ++ source_exclude ++ cover_exclude
+  exclude: line_exclude ++ erlang_doc_exclude ++ source_exclude ++ cover_exclude,
+  assert_receive_timeout: String.to_integer(System.get_env("ELIXIR_ASSERT_TIMEOUT", "300"))
 )
 
 defmodule IEx.Case do

--- a/lib/logger/test/logger/translator_test.exs
+++ b/lib/logger/test/logger/translator_test.exs
@@ -1275,7 +1275,7 @@ defmodule Logger.TranslatorTest do
   test "drops events if otp report handling is switched off" do
     {:ok, pid} = GenServer.start(MyGenServer, :ok)
     catch_exit(GenServer.call(pid, :error))
-    refute_receive {:event, _, _}, 200
+    refute_receive {:event, _, _}, 100
   end
 
   def task(parent, fun \\ fn -> raise "oops" end) do

--- a/lib/logger/test/test_helper.exs
+++ b/lib/logger/test/test_helper.exs
@@ -11,7 +11,8 @@ CoverageRecorder.maybe_record("logger")
 ExUnit.start(
   trace: !!System.get_env("TRACE"),
   include: line_include,
-  exclude: line_exclude
+  exclude: line_exclude,
+  assert_receive_timeout: String.to_integer(System.get_env("ELIXIR_ASSERT_TIMEOUT", "300"))
 )
 
 defmodule Logger.Case do

--- a/lib/mix/test/mix/dep_test.exs
+++ b/lib/mix/test/mix/dep_test.exs
@@ -946,7 +946,7 @@ defmodule Mix.DepTest do
 
           Mix.Tasks.Deps.Get.run([])
           Mix.Tasks.Deps.Compile.run([])
-          refute_receive {:mix_shell, :error, ["Could not compile :git_repo" <> _]}
+          refute_receive {:mix_shell, :error, ["Could not compile :git_repo" <> _]}, 100
         end)
       end)
     end

--- a/lib/mix/test/mix/umbrella_test.exs
+++ b/lib/mix/test/mix/umbrella_test.exs
@@ -375,7 +375,7 @@ defmodule Mix.UmbrellaTest do
 
         Mix.Task.run("compile", ["--verbose"])
         assert_receive {:mix_shell, :info, ["no compile bar"]}
-        refute_receive {:mix_shell, :info, ["Compiled lib/bar.ex"]}
+        refute_receive {:mix_shell, :info, ["Compiled lib/bar.ex"]}, 100
       end)
     end)
   end

--- a/lib/mix/test/test_helper.exs
+++ b/lib/mix/test/test_helper.exs
@@ -49,7 +49,8 @@ CoverageRecorder.maybe_record("mix")
 ExUnit.start(
   trace: !!System.get_env("TRACE"),
   exclude: epmd_exclude ++ os_exclude ++ git_exclude ++ line_exclude ++ cover_exclude,
-  include: line_include
+  include: line_include,
+  assert_receive_timeout: String.to_integer(System.get_env("ELIXIR_ASSERT_TIMEOUT", "300"))
 )
 
 defmodule MixTest.Case do

